core/sail/memory/src/main/java/org/eclipse/rdf4j/sail/memory/MemorySailStore.java

```diff
@@ -569,13 +565,12 @@ public synchronized void deprecate(Resource subj, IRI pred, Value obj, Resource
 			}
 		}
 
-		private void acquireExclusiveTransactionLock()
+		private synchronized void acquireExclusiveTransactionLock()
```

Comment:
Private methods are rarely synchronized, as it is usually done on public/protected/package methods which then already have the monitor lock when calling into private methods. Is there a particular reason why this one needs to be synchronized or should we synchronize all of the methods that call it?